### ISSUE TITLE
osis support

### ISSUE THREAD

**Issue by pdattx** [digitalbiblesociety/browserbible-3#60]
I know the current support is only for usfm-files and bibles from unbound, but what do you think about the osis support? Is there any possibility to use a osis-bible? Or maybe in the next months?
PS: I wonder about the file "generate_osis" because it exists and in the package is the kjv2006 (in osis format) - but I always get the message "can not find: osis".

**Comment by johndyer** [comment]
I'm not sure why you're seeing the "can not find: osis"

I may try to refresh my installation to see if something is missing in the
live repo.

On Wed, Mar 11, 2015 at 4:34 PM, pdattx <notifications@github.com> wrote:

> I know the current support is only for usfm-files and bibles from unbound,
> but what do you think about the osis support? Is there any possibility to
> use a osis-bible? Or maybe in the next months?
> PS: I wonder about the file "generate_osis" because it exists and in the
> package is the kjv2006 (in osis format) - but I always get the message "can
> not find: osis".
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/digitalbiblesociety/browserbible-3/issues/60>.
>



-- 
John Dyer - http://j.hn/

**Comment by pdattx** [comment]
I downloaded the actual package and run the "node generate -a" command. The same with the "node generate -v osisbible" (where the generator is "osis" in the info.json).
There is a generate_osis.js in the node_modules-folder but when I try to generate this, I've got the same errormessage "Can't find: osis" on the commandline.

**Comment by pdattx** [comment]
You can see this Errormessage already by barryanders topic ( https://github.com/digitalbiblesociety/browserbible-3/issues/58#issuecomment-72991990 ).

**Comment by johndyer** [comment]
Yes, I'm guessing this actually is an error with jsdom and jquery
dependencies. The problem is that the OSIS builder needs jQuery which need
jsdom and jsdom has stopped doing updates for node.js.

So I need to do some work to remove jQuery and find an alternative...

JD

On Thu, Mar 12, 2015 at 2:44 PM, pdattx <notifications@github.com> wrote:

> You can see this Errormessage already by barryanders topic ( #58 (comment)
> <https://github.com/digitalbiblesociety/browserbible-3/issues/58#issuecomment-72991990>
> ).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/digitalbiblesociety/browserbible-3/issues/60#issuecomment-78589843>
> .
>



-- 
John Dyer - http://j.hn/

**Comment by pdattx** [comment]
Maybe with https://github.com/cheeriojs/cheerio ?

**Comment by pdattx** [comment]
Is there any other possibility to get an osis-bibleformat in this program? Maybe someone knows a good converter!?

**Comment by pdattx** [comment]
FYI: I found the Converter http://www.churchsw.org/2013/02/simple-bible-reader-converter.html , but sometimes you need to correct the export. Anyway, because the unbound-bible-format is the only one for the browserbible at this time, there are a big lost of information. All Titles, crossrefs,... are lost by the convert. As I see, is this the biggest restriction at this time - what is an browserbible without any full featured bibles.

**Comment by pdattx** [comment]
Does somebody know how we could reach a full-featured bible (with titles,croessrefs,...) in this actual version? I have really no more ideas to stop the lost of information - see my last comment.

**Comment by bennylin** [comment]
@pdattx have you found any solution to this?

**Comment by benjaminpick** [comment]
I had a similar error and fixed it by upgrading nodejs (>= v4 is needed).

**Comment by pdattx** [comment]
Yes, I found a big help for me: Please look at this converter, that's the best what I found out to handle all the different formats. And the developer is very helpful, even if you have suggestions for his work.
https://github.com/schierlm/BibleMultiConverter/releases
